sled-agent/src/bootstrap/agent.rs

```diff
@@ -965,40 +976,34 @@ impl Agent {
     }
 }
 
-// We intentionally DO NOT derive `Debug` or `Serialize`; both provide avenues
+// We intentionally DO NOT derive `Debug``; it provides avenues
 // by which we may accidentally log the contents of our trust quorum share.
-#[derive(Deserialize, PartialEq)]
+#[derive(Clone, Serialize, Deserialize, PartialEq)]
 struct PersistentSledAgentRequest<'a> {
     request: Cow<'a, SledAgentRequest>,
     trust_quorum_share: Option<SerializableShareDistribution>,
 }
 
-impl PersistentSledAgentRequest<'_> {
-    /// On success, the returned string will contain our raw
-    /// `trust_quorum_share`. This method is named `danger_*` to remind the
-    /// caller that they must not log this string.
-    fn danger_serialize_as_toml(&self) -> Result<String, toml::ser::Error> {
-        #[derive(Serialize)]
-        #[serde(remote = "PersistentSledAgentRequest")]
-        struct PersistentSledAgentRequestDef<'a> {
-            request: Cow<'a, SledAgentRequest>,
-            trust_quorum_share: Option<SerializableShareDistribution>,
-        }
-
-        let mut out = String::with_capacity(128);
-        let serializer = toml::Serializer::new(&mut out);
-        PersistentSledAgentRequestDef::serialize(self, serializer)?;
-        Ok(out)
+impl<'a> Ledgerable for PersistentSledAgentRequest<'a> {
+    fn is_newer_than(&self, _other: &Self) -> bool {
+        true
```

Comment:
@jgallagher Would appreciate your input here.

This is a type which we really need to serialize and deserialize to/from TOML to put it in persistent storage.

I'm using a utility called `Ledger` to help out with the "write back to both M.2s, cope if only one is actually on-line" logic".

This utility requires that `Ledgerable` types implement `Serialize` and `Deserialize`, which makes this type a pain-in-the-butt, since it wasn't previously Serializable.

Reply:
I don't think we want to store trust quorum shares in the ledger, but since we don't really have trust quorum shares anyway... How painful would it be to remove the `trust_quorum_share` field from `PersistentSledAgentRequest` altogether?

Reply:
I thought this was where we *should* store trust quorum shares when they're ready though? I can rip this out, but I'd also probably ripping out a lot more of the foundation of where trust quorum shares are stored. Can give this a shot in a follow-up commit.

Reply:
From chat:

If we pull on this thread and don't store tq shares, then a major portion of the bootstrap server also needs to be re-worked. This may or may not be what we wanna do, but it should probably be decoupled from this PR either way.

Reply:
I mentioned this to Sean in chat, but I think it's fine to move everything including keyshares over to the ledger right now. The protocol for TQ is going to change a bunch and will require a bunch of work in the bootstrap server as mentioned. It makes sense to decouple to a separate PR.